### PR TITLE
Use templateSrv to interpolate timeFrom and timeTo variables.

### PR DESCRIPTION
**Pull request by carvid**
I realized that `timeFrom` and `timeTo` variables can be used to access the overridden time range values at the panel level (Query Options). Those variables were interpolated using a regex instead of using `templateSrv`, thus preventing custom date formats to be used. With this change, it's now possible to specify a custom date format for `timeFrom` and `timeTo` variables if running on Grafana >= v7.2. For example:

```graphql
query {
  data:submissions(
    where: {
      submitTime: {_gte: "${timeFrom:date}", _lte: "${timeTo:date}"},
    }
  ) {
    Time:submitTime
  }
}
```

### CODE REVIEW

**Comment by retrodaredevil** [comment]
I meant to make this PR a long time ago but I never got around to it. It looks good to me, we might consider removing timeFrom and timeTo altogether unless we want to keep backwards compatibility with older queries since we just use `__from` and `__to` now.

But this is definitely much better than what we had before.

**Comment by retzkek** [comment]
Date formats are fully supported using the builtin  `$__from` and `$__to` variables, which are the recommended/documented variables to use now. I do think we should keep the legacy variables for backwards comparability, but I don't think we should add new features to them.

**Comment by carvid** [comment]
@retrodaredevil and @retzkek I might getting this wrong since I'm not a Grafana expert, but Isn't `timeFrom` and `timeTo` meant to access the overridden time range from the panel's "Query Options"?. I'm asking that because the builtin `$__from` and `$__to` always return the dashboard's time range.

**Comment by retzkek** [comment]
> isn't timeFrom and timeTo meant to access the overridden time range from the panel's "Query Options"?.

No, they were just internal variables, from before the global builtins were added in Grafana 7.

>  the builtin $__from and $__to always return the dashboard's time range.

It looks like that's a bug https://github.com/grafana/grafana/issues/20147

**Comment by carvid** [comment]
@retzkek thanks for the clarifying that. It seems that bug is there since Grafana v6.4.3. Other datasource plugins provide a different ways to access the time range so I guess that's the reason the bug is not fixed yet (e.g. MySQL and Postgres datasource provide `__timeFilter`, `__timeFrom` and `__timeTo` among other macros).

For now, I guess `timeFrom` and `timeTo` are the only workaround to get the panel's relative time for this plugin.

File: src/DataSource.ts [hunk 1]
```diff
@@ -70,8 +70,10 @@ export class DataSource extends DataSourceApi<MyQuery, MyDataSourceOptions> {
   private createQuery(query: MyQuery, range: TimeRange | undefined, scopedVars: ScopedVars | undefined = undefined) {
     let payload = query.queryText;
     if (range) {
-      payload = payload.replace(/\$timeFrom/g, range.from.valueOf().toString());
-      payload = payload.replace(/\$timeTo/g, range.to.valueOf().toString());
+      payload = getTemplateSrv().replace(payload, {
+        timeFrom: { text: 'from', value: range.from.valueOf() },
```
Comment (retzkek) [inline src/DataSource.ts:74]: What do the `text` values mean? Does it matter what they're set to?

**Review by retzkek — COMMENTED** [review]
I've tested this, sorry for the delay. It seems to work OK, although unfortunately the `date` format options that [are provided for `__from` and `__to`](https://grafana.com/docs/grafana/latest/variables/variable-types/global-variables/#__from-and-__to) don't seem to work, but the behavior doesn't change from what we had before, so :shrug:. Presumably this is some limitation of `templateSrv`, but documentation here is scarce.

File: src/DataSource.ts (hunk 1 above)
Comment (carvid) [reply src/DataSource.ts:74]: To be honest, I don't know what the `text` property is and what it is used for. It is required by `ScopedVar` interface, though (https://github.com/grafana/grafana/blob/7520ebadacf2f5276a1f187f1bf4cfc4b953ab23/packages/grafana-data/src/types/ScopedVars.ts#L1). I didn't spend too much time selecting a good value for that property, but I can change it for a more meaningful message. Any suggestions?.

Comment (carvid) [reply src/DataSource.ts:74]: As for the `date` format options, they worked just fine for me with `timeFrom` and `timeTo`:

Screen Shot 2020-10-30 at 8 56 09 PM

What version of Grafana did you use for your tests?. I think the `date` format options were introduced in Grafana v7.2: https://grafana.com/blog/2020/09/23/grafana-v7.2-released-with-custom-date-formats-new-transforms-and-overrides/

Comment (retzkek) [reply src/DataSource.ts:74]: >  I can change it for a more meaningful message

Nah, it's fine, I was having trouble figuring out what it was used for myself so just was curious if you knew. :)

> As for the date format options, they worked just fine for me with timeFrom and timeTo

You're right, sorry. It seems while I was running 7.2.2 backend I was still getting some old frontend code.

**Comment by retzkek** [comment]
LGMT. Thanks for contributing!

**Pull request merged**